libs/knowledge-store/tests/integration_tests/test_graph_store.py

```diff
@@ -0,0 +1,79 @@
+import secrets
+from typing import Iterator
+from langchain_openai import OpenAIEmbeddings
+import pytest
+from testcontainers.core.container import DockerContainer
+from testcontainers.core.waiting_utils import wait_for_logs
+from cassandra.cluster import Cluster
+from dotenv import load_dotenv
+
+from ragstack_knowledge_store.graph_store import GraphStore
+
+load_dotenv()
+
+KEYSPACE = "default_keyspace"
+
+@pytest.fixture(scope="session")
+def cassandra_port() -> Iterator[int]:
+    # TODO: Allow running against local Cassandra and/or Astra using pytest option.
+    cassandra = DockerContainer("cassandra:5")
+    cassandra.with_exposed_ports(9042)
+    cassandra.with_env(
+        "JVM_OPTS",
+        "-Dcassandra.skip_wait_for_gossip_to_settle=0 -Dcassandra.initial_token=0",
+    )
+    cassandra.with_env("HEAP_NEWSIZE", "128M")
+    cassandra.with_env("MAX_HEAP_SIZE", "1024M")
+    cassandra.with_env("CASSANDRA_ENDPOINT_SNITCH", "GossipingPropertyFileSnitch")
+    cassandra.with_env("CASSANDRA_DC", "datacenter1")
+    cassandra.start()
+    wait_for_logs(cassandra, "Startup complete")
+    cassandra.get_wrapped_container().exec_run(
+        (
+            f"""cqlsh -e "CREATE KEYSPACE {KEYSPACE} WITH replication = """
+            '''{'class': 'SimpleStrategy', 'replication_factor': '1'};"'''
+        )
+    )
+    port = cassandra.get_exposed_port(9042)
+    print(f"Cassandra started. Port is {port}")
+    yield port
+    cassandra.stop()
+
+@pytest.fixture
+def graph_store_factory(cassandra_port: int):
+    tables = []
+
+    cluster = Cluster(
+        port=cassandra_port,
+    )
+    session = cluster.connect()
+    session.set_keyspace(KEYSPACE)
+
+    embedding = OpenAIEmbeddings()
+
+    def _make_graph_store():
+        name = secrets.token_hex(8)
+
+        node_table = f"nodes_{name}"
+        targets_table = f"targets_{name}"
+        return GraphStore(
+            embedding,
+            session = session,
+            keyspace = KEYSPACE,
+            node_table = node_table,
+            targets_table = targets_table,
+        )
+
+    yield _make_graph_store
+
+    for table in tables:
+        session.execute(f"DROP TABLE IF EXISTS {table}")
```

Comment:
Arguably, since these only run against a cassandra docker image, I could skip dropping things for even faster tests. Thoughts?

Reply:
yes you could indeed. Would be great to refactor this to depends on ragstack-test-utils which already have the container code

Reply:
I would, but I'm worried that will lead to the same problems I have with `libs/langchain/tests/integration_tests`. Specifically, I can't run those locally because something just hangs forever about starting up the container. This works quickly and is easy to interact with since it's all in one file. I generally prefer tests being straightforward, and this seems to simplify things (keeping it in one place) rather than entangling with other utilities...